mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,660 @@
+        return positive_props
+
+    def get_negatives(self,
+                      proposals,
+                      incomplete_iou_thresh,
+                      bg_iou_thresh,
+                      bg_coverage_thresh=0.01,
+                      incomplete_overlap_thresh=0.7):
+        """Get negative proposals, including incomplete proposals and
+        background proposals.
+
+        Args:
+            proposals (list): List of proposal instances.
+            incomplete_iou_thresh (float): Maximum threshold of overlap
+                of incomplete proposals and groundtruths.
+            bg_iou_thresh (float): Maximum threshold of overlap
+                of background proposals and groundtruths.
+            bg_coverage_thresh (float): Minimum coverage of background
+                proposals in video duration.
+            incomplete_overlap_thresh (float): Minimum percent of
+                incomplete proposals' own span contained in a
+                groundtruth instance.
+        """
+        incomplete_props = []
+        background_props = []
+
+        for i in range(len(proposals)):
+            if (proposals[i].best_iou < incomplete_iou_thresh
+                    and proposals[i].overlap_self > incomplete_overlap_thresh):
+                incomplete_props.append(proposals[i])
+            elif (proposals[i].best_iou < bg_iou_thresh
+                  and proposals[i].coverage > bg_coverage_thresh):
+                background_props.append(proposals[i])
+
+        return incomplete_props, background_props
+
+    def _video_centric_sampling(self, record):
+        """Sample proposals from the this video instance.
+
+        Args:
+            record (dict): Information of the video instance.
+        """
+        fg = self.get_positives(record['gts'], record['proposals'],
+                                self.assigner.fg_iou_thr,
+                                self.sampler.add_gt_as_proposals)
+        incomp, bg = self.get_negatives(record['proposals'],
+                                        self.assigner.incomplete_iou_thr,
+                                        self.assigner.bg_iou_thr,
+                                        self.assigner.bg_coverage_thr,
+                                        self.assigner.incomplete_overlap_thr)
+
+        def sample_video_proposals(proposal_type, video_id, video_pool,
+                                   requested_num, dataset_pool):
+            """This method will sample proposals from the this video pool. If
+            the video pool is empty, it will fetch from the dataset pool
+            (collect proposal of the entire dataset).
+
+            Args:
+                proposal_type (int): Type id of proposal.
+                    Positive/Foreground: 0
+                    Negative:
+                        Incomplete: 1
+                        Background: 2
+                video_id (str): Name of the video.
+                video_pool (list): Proposals in this video.
```

Comment:
so what dose the `pool` mean?

Reply:
The pool comprised of proposals in this video. I will revised the docstring.